File: ecommerce-frontend/src/services/paymentService.ts
import api from "./api";

export interface PaymentOrderResponse {
  orderId: number;
  razorpayOrderId: string;
  amount: number;
  currency: string;
  keyId: string;
}

export interface PaymentVerificationRequest {
  orderId: number;
  razorpayOrderId: string;
  razorpayPaymentId: string;
  razorpaySignature: string;
}

export const createPaymentOrder = async(
    orderId:number
):Promise<PaymentOrderResponse> =>{
    const response = await api.post<PaymentOrderResponse>(
        `/payments/create-order?orderId=${orderId}`
    )

    return response.data;
}

export const verifyPayment = async (
  data: PaymentVerificationRequest
): Promise<void> => {
  await api.post("/payments/verify", data);
};

File: src/main/java/com/harsh/ecommerce/service/PaymentService.java
package com.harsh.ecommerce.service;

import com.harsh.ecommerce.dto.request.PaymentVerificationRequest;
import com.harsh.ecommerce.dto.response.PaymentOrderResponse;

public interface PaymentService {

     PaymentOrderResponse createPaymentOrder(Long orderId);

     void verifyPayment(PaymentVerificationRequest request);
}